src/org/zaproxy/zap/extension/ruleconfig/ExtensionRuleConfig.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2016 The ZAP Development team
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ruleconfig;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.zaproxy.zap.extension.api.API;
+
+public class ExtensionRuleConfig extends ExtensionAdaptor {
+
+	public static final String NAME = "ExtensionRuleConfig";
+	
+	private OptionsRuleConfigPanel optionsRuleConfigPanel;
+	private RuleConfigParam ruleConfigParam;
+
+	public ExtensionRuleConfig() {
+		super();

```

Comment:
Could use the constructor `ExtensionAdaptor(String)` with `NAME` as parameter (`initialize()` could be removed).